### ISSUE TITLE
Port to Mac using the new tools introduced by apple

### ISSUE THREAD

**Issue by Woodentables** [beyond-all-reason/Beyond-All-Reason#2008]
Just wondering if BAR could possibly be ported to macOS using apple's guide and the tools they've provided. This guide can be found here: https://developer.apple.com/videos/play/wwdc2023/10123/.

**Comment by sprunk** [comment]
https://github.com/beyond-all-reason/spring/issues/936 possibly relevant

**Comment by Karolson** [comment]
As far as I know Apple's toolkit is just a fancy Wine wrapper like Proton/Crossover, with additional proprietary DirectX->Metal translator based on DXVK, and it won't help with OpenGL

**Comment by alanpaone** [comment]
yeah, Apple's tools only help with directX games. currently, wine (which is what crossover and Game Porting Toolkit are both based on) only supports opengl 2.1 . you'd be waiting for crossover to integrate [MGL](https://github.com/openglonmetal/MGL), Zink, or roll their own openGL 4.6 support somehow. Asahi linux has opengl 3.3 support, and at the breakneck pace they're going, it might eventually be the easiest way to run BAR on a mac, but it isn't yet. In theory, vmware Fusion 13 with an Arm Linux guest has opengl 4.6 support, but it [can't get past the launcher](https://github.com/beyond-all-reason/Beyond-All-Reason/issues/2020).   

 If you can put up with some UI elements and explosions not rendering correctly, BAR runs decently in Parallels.

**Comment by alanpaone** [comment]
Parallels 19 just came out and adds opengl 4.1, it still doesn't show a lot of the UI, so I didn't do a lot of testing, but I wouldn't be surprised if it performs or looks slightly better than parallels 18. Also, vmware Fusion tech preview 2023 doesn't work with opengl on windows, but does have UI elements that mention 4.3 (along with other notes elsewhere that specifically say they're wrong), so it looks like a future release will include 4.3 on windows. so close yet so far.

**Comment by mustafa-hanif** [comment]
if its an open source game why cant we compile it to mac

**Comment by p2004a** [comment]
> if its an open source game why cant we compile it to mac

Because Apple doesn't support OpenGL 4.6 and new macs are using ARM not AMD64 architecture. We can't just compile it, it requires porting work both on the CPU architecture side and the Graphics API side.

**Comment by mustafa-hanif** [comment]
I am willing to contribute to the effort of porting it

**Comment by p2004a** [comment]
@mustafa-hanif Great! Engine repository is https://github.com/beyond-all-reason/spring it contains compilation instructions and so on. Best if you joined our Discord for asking questions, but in the readme there is also linked bridged Matrix channel if you don't want to use Discord. Nobody is working in this area at the moment, so you can start however you want.

**Comment by tobyhinloopen** [comment]
@mustafa-hanif did you end up looking at the possibilities?

**Comment by maxmoros** [comment]
@mustafa-hanif Did you make any progress on this? I'm considering taking a look as well.

**Comment by czekaj** [comment]
We should just ask @claude to do it, lol

**Comment by sprunk** [comment]
Sure, see the attached engine ticket (people are already vibe coding it and already found suckers willing to hunt for desyncs).
